packages/features/ee/teams/components/TeamEventTypeForm.tsx

```diff
@@ -0,0 +1,159 @@
+import type { EventType } from "@prisma/client";
+import type { ReactNode } from "react";
+
+import { useOrgBranding } from "@calcom/features/ee/organizations/context/provider";
+import { classNames } from "@calcom/lib";
+import { useCreateEventType } from "@calcom/lib/hooks/useCreateEventType";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import slugify from "@calcom/lib/slugify";
+import { SchedulingType } from "@calcom/prisma/enums";
+import { trpc } from "@calcom/trpc/react";
+import { Form, TextField, Tooltip } from "@calcom/ui";
+import { Alert, RadioGroup as RadioArea } from "@calcom/ui";
+
+type props = {
+  isTeamAdminOrOwner: boolean;
+  teamId: number;
+  SubmitButton: (isPending: boolean) => ReactNode;
+  onSuccessMutation: (eventType: EventType) => void;
+  onErrorMutation: (message: string) => void;
+  isInfiniteScrollEnabled: boolean;
+};
+export const TeamEventTypeForm = ({
+  isTeamAdminOrOwner,
+  teamId,
+  SubmitButton,
+  onSuccessMutation,
+  onErrorMutation,
+  isInfiniteScrollEnabled,
+}: props) => {
+  const { t } = useLocale();
+  const orgBranding = useOrgBranding();
+  const { data: team } = trpc.viewer.teams.get.useQuery({ teamId, isOrg: false }, { enabled: !!teamId });
```

Comment:
Can we wait for my PR to get merged first https://github.com/calcom/cal.com/pull/16155 and then use `viewer.teams.getTeamWithMinimalData` instead because `viewer.teams.get` wouldn't work for big teams and take too much time to load?

Reply:
No issue; that is a higher priority than this one